### ISSUE TITLE
Does the API support a way to enumerate connected devices?

### ISSUE THREAD

**Issue by ciaran2** [jkuhlmann/gainput#33]
Multiple devices can be registered with the `InputManager` by means of repeated calls to `CreateDevice()`.  I don't see any clear way to verify that a device is in fact a real connected device, though. It seems like no matter how many devices I register of any type (significantly more than are actually present), the variant of the registered device is always DV_STANDARD, never DV_NULL.

Am I thinking about this completely the wrong way, or is there something to check device validity that I'm not seeing, or is this genuinely a missing feature?

**Comment by jkuhlmann** [comment]
You should do it this way: Create all the devices you want to possibly support. Then, if you need to know this, you can call InputDevice::IsAvailable() or InputDevice::GetState() to determine what state the device is in. 

Does that help?

**Comment by ciaran2** [comment]
It does, sort of.  Now I'm getting the same `DeviceState` for all devices of the same type.  Does one device object of a given type govern all devices of that type?

**Comment by jkuhlmann** [comment]
No, it doesn't. Though you may run into limitations as, for example, multiple keyboards are mice are not supported. It should work for gamepads though. What device type are we talking about?

**Comment by ciaran2** [comment]
What I'm seeing, after creating five pads, three keyboards, and three mice, with one of each device actually connected, is that all of the keyboards and mice are `DS_OK` and all of the pads are `DS_UNAVAILABLE`.

I could have sworn that the pad I was using to test was XInput compatible, but assuming I'm wrong and it's not, is that that the output you would expect to see?

**Comment by jkuhlmann** [comment]
Yes, assuming the pad doesn't work with Xinput that's exactly what I'd expect. 
What gamepad is it? Do you have any other way to test if it's supported?

**Comment by ciaran2** [comment]
I have yet to find an easy, conclusive test for Xinput compatibility.  The
pad is a Logitech DualAction.

**Comment by jkuhlmann** [comment]
Googling around a little, it seems like the Logitech Dual Action is not directly supported by Xinput, but there are 3rd party drivers/emulators to make it compatible.
